### PR TITLE
Let wcom execute check-manager-config query.

### LINKED ISSUE #8356: Avoid execd to execute restart_manager and check_manager_config queries
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| all | execd | Managert | Packages/Sources | All |

Hi team.

At this moment, `execd` is the daemon that is processing the `restart_manager` and `check_manager_config` queries. 
This has problems when active response is disabled, as the `execd` daemon is not started and the `restart_manager` and `check-manager-config` API endpoints fail. 

This issue aims to change these queries to the `wcom` thread, as this thread should be the one in charge of doing these queries.

To replicate the API behavior, I have used the following python script  (the code of this script is taken from the API code):
```python
import socket
import json

from os import remove
from os.path import exists, join
import struct
import sys

execq_socket_path = '/var/ossec/queue/alerts/execq'


def parse_execd_output(output: str):
    json_output = json.loads(output)
    return json_output


def create_wazuh_socket_message(origin=None, command=None, parameters=None):
    communication_protocol_message = {'version': 1}

    if origin:
        communication_protocol_message['origin'] = origin

    if command:
        communication_protocol_message['command'] = command

    if parameters:
        communication_protocol_message['parameters'] = parameters

    return communication_protocol_message


def test_exec():
    """Check if Wazuh configuration is OK.
    Returns
    -------
    response : str
        Status of the configuration.
    """
    try:
        # Sockets path
        api_socket_relative_path = join('queue', 'alerts', 'execa')
        api_socket_path = join('/var/ossec', api_socket_relative_path)
        # Message for checking Wazuh configuration

        execq_msg = json.dumps(create_wazuh_socket_message(origin={'module': 'api/framework'},
                                                           command='check-manager-configuration',
                                                           parameters={"extra_args": [], "alert": {}}))
        # Remove api_socket if exists
        try:
            remove(api_socket_path)
        except OSError as e:
            if exists(api_socket_path):
                raise e

        # up API socket
        try:
            api_socket = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
            api_socket.bind(api_socket_path)
            # Timeout
            api_socket.settimeout(10)
        except OSError as e:
            raise e

        # Connect to execq socket
        if exists(execq_socket_path):
            try:
                execq_socket = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
                execq_socket.connect(execq_socket_path)
            except OSError as e:
                raise e
        else:
            exit(1)
        # Send msg to execq socket
        try:
            execq_socket.send(execq_msg.encode())
            execq_socket.close()
        except socket.error as e:
            raise e
        finally:
            execq_socket.close()

        # If api_socket receives a message, configuration is OK
        try:
            buffer = bytearray()
            # Receive data
            datagram = api_socket.recv(4096)
            buffer.extend(datagram)
        except socket.timeout as e:
            raise e
        finally:
            api_socket.close()
            # Remove api_socket
            if exists(api_socket_path):
                remove(api_socket_path)

        try:
            response = parse_execd_output(buffer.decode('utf-8').rstrip('\0'))
        except (KeyError, json.decoder.JSONDecodeError) as e:
            raise e
    finally:
        pass
    return response


def test_com():
    """Check if Wazuh configuration is OK.
    Returns
    -------
    response : str
        Status of the configuration.
    """
    # Sockets path
    api_socket_relative_path = join('/var', 'ossec', 'queue', 'sockets', 'com')
    print(api_socket_relative_path)
    command = 'check-manager-config'
    # Message for checking Wazuh configuration
    con = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
    con.connect(api_socket_relative_path)
    encoded_msg = command.encode('utf-8')
    con.send(struct.pack("<I", len(encoded_msg)) + encoded_msg)
    size = struct.unpack("<I", con.recv(4, socket.MSG_WAITALL))[0]
    recv_msg = con.recv(size, socket.MSG_WAITALL)
    con.close()
    return recv_msg


if __name__ == '__main__':
    if len(sys.argv) != 2:
        print('Wrong number of arguments.')
        exit(1)

    if sys.argv[1] == 'w':
        response = test_com()
    else:
        response = test_exec()

    print(response)
```
If the script is executed using the `e` argument, it will try to connect to the `execq` socket. If the argument is different, it will connect to the `wcom` socket emulating the API behavior.

If active response is disabled, the script will raise an exception, as it can't connect to the `execq` socket.

### Configuration options
```xml
<active-response>
  <disabled>yes</disabled>
  <command>firewall-drop</command>
  <location>local</location>
</active-response>
```
```
root@ubuntumanager:/vagrant/Scripts# python3 test_execq.py e
Traceback (most recent call last):
  File "test_execq.py", line 135, in <module>
    response = test_exec()
  File "test_execq.py", line 70, in test_exec
    raise e
  File "test_execq.py", line 68, in test_exec
    execq_socket.connect(execq_socket_path)
ConnectionRefusedError: [Errno 111] Connection refused
```

### PR DESCRIPTION
**Pull request by antoniomanuelfr**
|Related issue|
|---|
|#8356|

## Description
This PR aims to avoid `execd` to execute `check-manager-config` and `restart-manager` API requests. Now, these requests that were executed by `execd` will be executed by the `wcom` thread. This will avoid the problems described in #7968 once this development is completed.

If the `check-manager-config` request is sent to execd after the change, the following error will appear:
```
2021/04/26 06:57:29 wazuh-execd: ERROR: (1311): Invalid command name 'check-manager-configuration' provided.
```
Closes #8356
## Test active response disabled
```
root@ubuntumanager:/vagrant/Scripts# python3 test_execq.py e
Traceback (most recent call last):
  File "test_execq.py", line 135, in <module>
    response = test_exec()
  File "test_execq.py", line 89, in test_exec
    raise e
  File "test_execq.py", line 86, in test_exec
    datagram = api_socket.recv(4096)
socket.timeout: timed out
root@ubuntumanager:/vagrant/Scripts# python3 test_execq.py w
/var/ossec/queue/sockets/com
b'{"error":0,"message":"ok"}'
root@ubuntumanager:/vagrant/Scripts# 
```
## Test active response enabled
```
root@ubuntumanager:/vagrant/Scripts# python3 test_execq.py e
Traceback (most recent call last):
  File "test_execq.py", line 135, in <module>
    response = test_exec()
  File "test_execq.py", line 89, in test_exec
    raise e
  File "test_execq.py", line 86, in test_exec
    datagram = api_socket.recv(4096)
socket.timeout: timed out
root@ubuntumanager:/vagrant/Scripts# python3 test_execq.py w
/var/ossec/queue/sockets/com
b'{"error":0,"message":"ok"}'
root@ubuntumanager:/vagrant/Scripts# 
```
The exception that is raised is because the API was expecting to connect to the `alerts/execa` sockets, which is no longer used and it won't be created.
## Valgrind report 
```
==29915== LEAK SUMMARY:
==29915==    definitely lost: 0 bytes in 0 blocks
==29915==    indirectly lost: 0 bytes in 0 blocks
==29915==      possibly lost: 272 bytes in 1 blocks
==29915==    still reachable: 144 bytes in 1 blocks
==29915==         suppressed: 0 bytes in 0 blocks
```

- Compilation without warnings in every supported platform
  - [X] Linux
- [X] Source installation
- [x] Source upgrade

- Memory tests for Linux
  - [x] Scan-build report
  - [ ] Coverity
  - [X] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

File: src/os_execd/wcom.c [hunk 1]
```diff
@@ -304,6 +307,73 @@ size_t wcom_getconfig(const char * section, char ** output) {
...
+        if (wm_exec(command_in, &output_msg, &result_code, 2000, NULL) < 0) {
+            if (result_code == EXECVE_ERROR) {
+                mwarn("Path is invalid or file has insufficient permissions. %s", command_in);
+            } else {
+                mwarn("Error executing [%s]", command_in);
+            }
+
+            ret_val = 1;
+            size_t size = snprintf(NULL, 0, "Error executing %s - (%d)", command_in, result_code);
+            os_calloc(size + 1, sizeof(char), msg);
```
Comment (Molter73) [inline src/os_execd/wcom.c:338]: What happens if the for loop has run once or twice before failing and calling this `os_calloc`? I believe the `wm_strcat`call in line 346 could have allocated memory and stored it in `msg`, making this call a memory leak.

We should try to rethink how we handle this case, memory leak aside, it's a little messy and hard to follow.

**Review by Molter73 — CHANGES_REQUESTED** [review]

**Review by Molter73 — DISMISSED** [review]
LGTM!

**Review by Molter73 — DISMISSED** [review]

**Review by vikman90 — APPROVED** [review]
